src/plugins/ssh/index.ts

```diff
@@ -345,60 +290,79 @@ const transformForShell = (args: string[]) => {
   });
 };
 
-const awsLogin = async (authn: Authn, data: AwsSshRequest) => {
-  if (!(await ensureSsmInstall())) {
-    throw "Please try again after installing the required AWS utilities";
-  }
-
-  const { config } = await getAwsConfig(authn, data.accountId);
-  if (!config.login?.type || config.login?.type === "iam") {
-    throw "This account is not configured for SSH access via the P0 CLI";
+/** Construct another command to use for testing access propagation prior to actually logging in the user to the ssh session */
+const preTestAccessPropagationIfNeeded = async (
+  request: SshRequest,
+  cmdArgs: CommandArgs,
+  proxyCommand: string[],
+  credential: AwsCredentials | undefined
+) => {
```

Comment:
We can improve the typing here, and fully remove the dependency on AWS in the generic SSH code:

```suggestion
const preTestAccessPropagationIfNeeded = async <P extends SshProvider<any, any, any, any>>(
  request: SshRequest,
  cmdArgs: CommandArgs,
  proxyCommand: string[],
  credential: P extends SshProvider<infer _PR, infer _O, infer _SR, infer C> ? C : undefined
) => {
```

Reply:
Nice!